lib/utils/browserSupportedFormats.ts

```diff
@@ -0,0 +1,33 @@
+import { Format } from '../types'
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/API/MediaSource
+ * Most browsers supports this, except Safari on iPhone
+ */
+const MSE_SUPPORT = 'MediaSource' in window
+
+const isMJPEGSupported = () => {
+  const isInternetExplorer =
+    navigator.userAgent.includes('MSIE') || 'ActiveXObject' in window
+  return !isInternetExplorer
+}
+
+const isH264Supported = () => {
+  const element = document.createElement('video')
+
+  // Return type specified here https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/canPlayType
+  const type = element.canPlayType('video/mp4; codecs="avc1.640029"') as
+    | 'probably'
+    | 'maybe'
+    | ''
```

Comment:
This should already be typed as `CanPlayTypeResult` by typescript, why do you need `as` here?

Reply:
Done